diff --git modules/simpletest/tests/schema.test modules/simpletest/tests/schema.test
index 8945117..c3fad57 100644
--- modules/simpletest/tests/schema.test
+++ modules/simpletest/tests/schema.test
@@ -125,6 +125,7 @@ class SchemaTestCase extends DrupalWebTestCase {
           'mysql_type' => 'timestamp',
           'pgsql_type' => 'timestamp',
           'sqlite_type' => 'datetime',
+          'oci_type' => 'timestamp',
           'not null' => FALSE,
           'default' => NULL,
         ),
